### PR TITLE
Survey

### PR DESCRIPTION
**Pull request by peterlogg**
Working version of survey backend routes. All tests run successfully but need access to instantiated postgres DB instance at present.

### CODE REVIEW

File: src/main/scala/com/focaldata/cinprojectsurvey/repositories/surveys/Surveys.scala [hunk 1]
```diff
@@ -20,34 +25,81 @@ trait Surveys {
...
-    def updateState(survey: Survey) = sql"""update survey set sstate=${survey.state}::sstate where id=${survey.id} returning *;""".query[Survey]
+    def querySurvey(surveyIdInput: SurveyIdInput) = {
+      sql"""
+      select survey_id,
+      nickname,
+      name,
+      state,
+      created_date,
+      since_date
+      from survey where survey_id=${surveyIdInput};""".query[Survey]
```
Comment (tonymagne) [inline src/main/scala/com/focaldata/cinprojectsurvey/repositories/surveys/Surveys.scala]: I think you don't need the curly brackets when you directly use the argument
```suggestion
      from survey where survey_id=$surveyIdInput""".query[Survey]
```

File: src/main/scala/com/focaldata/cinprojectsurvey/repositories/surveys/Surveys.scala [hunk 2]
```diff
@@ -20,34 +25,81 @@ trait Surveys {
...
+      insert into survey (survey_id, nickname, name, state, created_date, since_date)
+        values (
+            ${survey.surveyId}, ${survey.nickname}, ${survey.name}, ${survey.state}::sstate,
+            ${survey.createdDate}, ${survey.sinceDate}) returning *;""".query[Survey]
 
+    def updateSurveyState(surveyUpdateStatusInput: SurveyUpdateStateInput, sinceDate: Instant) = sql"""
+      update survey set state=${surveyUpdateStatusInput.state}::sstate, since_date=${sinceDate} where survey_id=${surveyUpdateStatusInput.surveyId} returning *;""".query[Survey]
+
+    def deleteSurvey(surveyIdInput: SurveyIdInput) = sql"""
+      delete from survey where survey_id=$surveyIdInput returning *;""".query[Survey]
```
Comment (tonymagne) [inline src/main/scala/com/focaldata/cinprojectsurvey/repositories/surveys/Surveys.scala]: Oh, just did it here

File: src/test/scala/com/focaldata/cinprojectsurvey/repositories/SurveysSpec.scala [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+  val tsThen  = Instant.now()
+  val tsLater = Instant.now()
+
+  val surveyServiceUnderTest = new Surveys.Live {}
+  val allSurveysSuite = suite("Surveys module")(
+    testM("gets a survey by id") {
+      mkSurveyTestEnv.use { env =>
+        val testSurveyId = UUID.randomUUID
```
Comment (tonymagne) [inline src/test/scala/com/focaldata/cinprojectsurvey/repositories/SurveysSpec.scala]: Use `generateId` here so it is exercised a well

File: src/test/scala/com/focaldata/cinprojectsurvey/repositories/SurveysSpec.scala [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        val testSurveyInput = SurveyIdInput(testSurveyId)
+        val testSurvey = Survey(SurveyId(testSurveyId), Nickname("test_survey_2"), Name("Test survey 2"), SurveyState.draft, tsThen, tsLater)
```
Comment (tonymagne) [inline src/test/scala/com/focaldata/cinprojectsurvey/repositories/SurveysSpec.scala]: User `generateNickname ` here so it is exercised.

File: src/main/scala/com/focaldata/cinprojectsurvey/repositories/surveys/Surveys.scala (hunk 1 above)
Comment (peterlogg) [reply src/main/scala/com/focaldata/cinprojectsurvey/repositories/surveys/Surveys.scala]: changed

File: src/test/scala/com/focaldata/cinprojectsurvey/repositories/SurveysSpec.scala (hunk 3 above)
Comment (peterlogg) [reply src/test/scala/com/focaldata/cinprojectsurvey/repositories/SurveysSpec.scala]: done and throughout also

File: src/test/scala/com/focaldata/cinprojectsurvey/repositories/SurveysSpec.scala (hunk 4 above)
Comment (peterlogg) [reply src/test/scala/com/focaldata/cinprojectsurvey/repositories/SurveysSpec.scala]: done and throughout also

**Pull request merged**